File: doc/flexiReport/tools/makePVD/make_pvd.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

int main(int argc, char *argv[]){

	if(argc != 2){

		fprintf(stderr, "Usage: %s <file_increment>\n", argv[0]);
		exit(1);

	}

	bool exvar = true;
	FILE *test_file;
	FILE *output = fopen("einfeldt.pvd", "w");
    double nvar = 0;
	char filename[1000];

	double incr = atof(argv[1]);

	fprintf(output, "<VTKFile type=\"Collection\" version=\"1.0\" byte_order=\"LittleEndian\" header_type=\"UInt64\">\n");
    fprintf(output, "  <Collection>\n");

	
    

	do {

		sprintf(filename, "data/leblanc_Solution_%017.9f.vtm", nvar);

		test_file = fopen(filename, "r");

		if(!test_file) exvar = false;
		else {
			fclose(test_file);
		    printf("Writing file: %s\n", filename);
			/* write it into the pvd file */
            fprintf(output, "     <DataSet timestep=\"%f\" part=\"0\" file=\"%s\"/>\n", nvar, filename);
		    nvar += incr;
		}

	} while (exvar);

    fprintf(output, "  </Collection>\n</VTKFile>\n");


}

File: doc/flexiReport/tools/makePVD/makepvd.py
#!/usr/bin/env python3

import sys
import os
import argparse

#============================================
# setupParser: Setup command line argument parser
#============================================
def setupParser():
	'''Setup the argument parser and return the parser object'''

	parser = argparse.ArgumentParser()
	parser.add_argument('-i','--inFilePrefix',dest='vtmFile',
		                required=True,
		                help='prefix name of Solution .vtm file: e.g.,  testname from testname_Solution_0000000.000000000.vtm')
	parser.add_argument('-o','--out',dest='filename',
						default='vtm2pdv.pdv',
		                help='enter the output pvd file name')
	parser.add_argument('-t','--dt',dest='dt',
						required=True,
		                help='enter the time step between solution output files')
	parser.add_argument('-e','--extension',dest='ext',
		                default='vtm',
		                help='File extension: e.g.,  pvd from testname_Solution_0000000.000000000.pvd')
	return parser

#============================================
#                   MAIN
#============================================
if(__name__ == '__main__'):

	#--------------------------------
	# Handle arguments
	#--------------------------------
	parser = setupParser()
	args = parser.parse_args()

	# Extract parameters relevant to all problems
	inPrefix = args.vtmFile
	outExt = args.ext
	outFile = args.filename
	print("inPrefix: ", inPrefix)
	print("outFile: ", outFile)

	sys.stdout.write('  Writing to file: {:s}\n'.format(outFile))
	file = open(outFile, 'w+')

	file.write("<VTKFile type=\"Collection\" version=\"1.0\" byte_order=\"LittleEndian\" header_type=\"UInt64\">\n")
	file.write("  <Collection>\n")

	exvar = True
	time = 0.
	dt = float(args.dt) 

#==============================================
#   Setup input file name search
#==============================================

	while exvar:

		fname = '{:s}_Solution_{:0>17.9f}.{:s}'.format(inPrefix, time, outExt)

		if os.path.exists(fname):
			print("Writing file: {:s}\n".format(fname))

			# write file name into the pvd file 
			file.write("     <DataSet timestep=\"{:f}\" part=\"0\" file=\"{:s}\"/>\n".format( time, fname))
			time += dt;
		else:
			exvar = False

	file.write("  </Collection>\n</VTKFile>\n")
	file.close()

